asapdiscovery-ml/asapdiscovery/ml/dataset.py

```diff
@@ -357,14 +357,15 @@ def __init__(
         from dgllife.utils import SMILESToBigraph
 
         # Build dataframe
-        all_compound_ids, all_smiles, all_pic50, all_range, all_stderr = zip(
+        all_compound_ids, all_smiles, all_pic50, all_range, all_stderr, all_dates = zip(
```

Comment:
what if dates not present?

Reply:
I added `date_created` as an optional field in the `ExperimentalCompoundData` schema class, so any instance of the class will definitely have the field. it may be empty, but that shouldn't cause any issues here, and I think I've covered that case in the splitting methods